charts/tfy-karpenter-config/README.md

```diff
@@ -3,17 +3,18 @@ A Helm chart for Karpenter provisioners.
 
 ## Soci
 
-You can add [soci-snapshotter](https://github.com/awslabs/soci-snapshotter) in the nodes brought up by Karpenter using the `userData` field in the `AWSNodeTemplate` resource.
+You can add [soci-snapshotter](https://github.com/awslabs/soci-snapshotter) in the nodes brought up by Karpenter using the `userData` field in the `EC2NodeClass` resource.
 
 The script in `userData` does the following,
 
 1. Install and load the `fuse` kernel module.
-2. Downloads the binary from the `awslabs/soci-snapshotter` release page.
-3. Adds a configuration file for the `soci-snapshotter-grpc` finary.
+2. Download the binary from the `awslabs/soci-snapshotter` release page.
+3. Adds a configuration file for the `soci-snapshotter-grpc` binary.
 4. Adds a `systemd` service, which manages the `soci-snapshotter-grpc` process.
-5. Patches `containerd` config to use the `soci-snapshotter`.
+5. Patches `containerd` and `kubelet` config to use the `soci-snapshotter`.
 
-### `userData` script for `default` Node Template
+### `userData` script for `default` EC2 Node Class
+* Note that the following script assumes AL2023 and will **not** work on AL2.
 
 ```bash
```

Comment:
I was wondering, can we put this in files/
as `al2023-soci-userdata.sh` ?

Easier to maintain and we can add a special value in values.yaml which can inject this userdata on helm render

Reply:
Oh yes, I am developing like that only, will add in git.